### ISSUE TITLE
`@model` is very picky about what names it allows

### ISSUE THREAD

**Issue by Eilon** [rynowak/RazorPages#2]
For this code:

```
@model int
```

I get this error:

```
An error occurred during the compilation of a resource required to process this request. Please review the following specific error details and modify your source code appropriately.

Pages/Test.razor

need a type name
1.@model int
```

And for this code:

```
@model System.Int32
```

I get this error:

```
An error occurred during the compilation of a resource required to process this request. Please review the following specific error details and modify your source code appropriately.

'System' is a namespace but is used like a type
'System' is a namespace but is used like a type
'System' is a namespace but is used like a type
'System' is a namespace but is used like a type
```

**Comment by Eilon** [comment]
And perhaps it also requires that you **not** have a `.` in the model name.

**Comment by rynowak** [comment]
Fixed 8b289cfde9c0572a48e9d3be506bde51ad54ed09 and 308cece038380af1f7169d16d7fa672fd204529b
